### ISSUE TITLE
Missing upload to CTAN?

### ISSUE THREAD

**Issue by hpreusse** [plk/biblatex#1202]
Consider to upload biblatex (and biber) to CTAN, once you think it is time to release. Many thanks!

**Comment by plk** [comment]
Yes, that is the plan - just waiting for a missing biber build for FreeBSD.

**Comment by hpreusse** [comment]
Many thanks for response! So I'll be patient.

**Comment by moewew** [comment]
Maybe we need a way to communicate to people on Github and SourceForge that the branches/uploads are just preparations for the official release. Possibly we should rename the folders on SourceForge and merge git branches only after things have been sent off to CTAN.

@plk On a related note. I just realised both `.tar`s for `biblatex` 3.17 on SourceForge have a `PaxHeader` directories in the `doc` directories (and in `doc/examples`). I'm guessing these folders shouldn't be there.

**Comment by plk** [comment]
The problem is that the way the distribution is done for CTAN is to pull everything cleanly off Sourceforge once it's all there as that's where the external builders put their uploads.

I'll have a look at the PaxHeaders thing - that looks like a change in the OSX default `tar` program, again ...

**Comment by moewew** [comment]
Yeah, I know that SourceForge is also used for CTAN submission. I'm guessing what I was imagining is that while we wait for the binaries to come in we give the folders on SF a name that makes it sufficiently clear that thing are not released yet (and maybe hold off putting `biblatex` up and merging `dev` to `master` branches on GH as well) and that we only change the name to signify the official release once everything is up just before CTAN are notified. That would possibly make it slightly clearer to people that things are not officially released yet. (I think we have had issues like this in the past, can't find the links though.)

**Comment by plk** [comment]
We can do this, yes. When we have this problem, it's usually just because one builder (in this case FreeBSD) isn't contactable or has issues building otherwise it would be done very quickly.

**Comment by plk** [comment]
Can you check the biblatex upload TAR now? I want to see if GNU tar has the same issue. On OSX, it is not a visible issue as the built-in ignores the PAX metadata on untar too ...

**Comment by moewew** [comment]
The `PaxHeader`s are still there, I'm afraid.

I also noticed that the TDS.zip has a `biblatex.pdf` in `/doc` and one in `/doc/latex/biblatex`. I'm guessing the one in `/doc` shouldn't be there?

**Comment by plk** [comment]
I think the duplicate `biblatex.pdf` should be corrected although I'm not sure why that happened - I can't remember if there was a good reason as it was explicitly coded that way in the build script.

The PaX headers - I can't seem to replicate this - I've looked in the dirs you mention on OSX, Solaris and Linux using the current distribution tars and I don't see any PaX files in /doc or indeed anywhere - do you have a screenshot of what you see? I wonder if this your tar adding this on unpack? I checked to make sure that my tar wasn't using POSIX compat. Do you by any chance have the environment variable `POSIXLY_CORRECT` set?

**Comment by moewew** [comment]
Sorry, you are right, the `PaxHeader`s are gone. I downloaded the files again. Maybe I got a bad mirror when I checked a few minutes ago.

**Comment by plk** [comment]
Then it was the OSX12 default BSD tar which is defaulting to POSIX format. I switched to GNU tar which defaults to something more sensible.

**Comment by hpreusse** [comment]
> That would possibly make it slightly clearer to people that things are not officially released yet. (I think we have had issues like this in the past, can't find the links though.)

Many thanks for discussing this topic! For now I think it is purely my fault. In Debian we look for the released version on github: when the new version was tagged I simply packaged and uploaded it. I completely forgot about the dependency to biblatex. Hope I won't do the same mistake another time.

**Comment by moewew** [comment]
See also the confusion in https://github.com/Homebrew/homebrew-core/pull/93820.

**Comment by plk** [comment]
I forget that people use the `test.bcf` for that - it was only on SF as somewhere to put a test file but since people use this to verify new versions now, I have made a note in the release docs to update this on release.

**Comment by moewew** [comment]
Ah good catch. My point was rather that they seem to have pulled Biber from GitHub or SF and that some users seem to get their LaTeX packages (`biblatex`) from TeX live via CTAN, so that they would ship a Biber that is too new for current TeX live's `biblatex`.

**Comment by hpreusse** [comment]
AFAITC you uploaded the new release to CTAN. So from my point of view the issue could be closed.
